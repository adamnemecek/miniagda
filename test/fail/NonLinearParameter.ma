-- 2013-04-05

data Prod (A, B : Set) { pair (a : A) (b : B) }

data D (A : Set 1)
{ c : D (Prod A A) } -- not a pattern, currenlty not accepted
