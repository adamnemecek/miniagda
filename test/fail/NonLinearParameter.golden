MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "NonLinearParameter.ma" ---
--- scope checking ---
--- type checking ---
type  Prod : ^(A : Set) -> ^(B : Set) -> Set
term  Prod.pair : .[A : Set] -> .[B : Set] -> ^(a : A) -> ^(b : B) -> < Prod.pair a b : Prod A B >
term  a : .[A : Set] -> .[B : Set] -> (pair : Prod A B) -> A
{ a [A] [B] (Prod.pair #a #b) = #a
}
term  b : .[A : Set] -> .[B : Set] -> (pair : Prod A B) -> B
{ b [A] [B] (Prod.pair #a #b) = #b
}
type  D : ^(A : Set 1) -> Set
error during typechecking:
D
/// expected parameter to be a pattern, but I found [Prod A A]
